src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,84 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            Files.deleteIfExists(Path.of(toFileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot delete file " + toFileName, e);
+        }
+
+        String[] actionsList = readFile(fromFileName);
+        String report = createReport(actionsList);
+        writeReportToFile(report, toFileName);
+    }
+    
+    private String[] readFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String line = bufferedReader.readLine();
 
+            while (line != null) {
+                stringBuilder.append(line).append(System.lineSeparator());
+                line = bufferedReader.readLine();
+            }
+
+            return stringBuilder.toString().split("[\\n\\r\\t]+");
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can't find file", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
+        }
     }
+
+    private void writeReportToFile(String reportToWrite, String toFileName) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(
+                new FileWriter(toFileName, true))) {
```

Comment:
There's no need to open the file in append mode (using `new FileWriter(toFileName, true)`). Since you're already deleting the file if it exists before this method is called, you can simply use `new FileWriter(toFileName)`.